src/rdf4cpp/rdf/writer/SerializationState.cpp

```diff
@@ -0,0 +1,40 @@
+#include "SerializationState.hpp"
+
+namespace rdf4cpp::rdf::writer {
+
+class PrefixDataBuilder {
```

Comment:
I think this adds some unecessary complexity. I would prefer if you would just write out all prefixes manually using a chain `[..]TRY_WRITE()` in a for loop (inside `SerializationState::begin`.

This will only happen exactly once per however many billion triples so the minor perf benefit will never show.